sentry-android-replay/src/main/java/io/sentry/android/replay/screenshot/CanvasStrategy.kt

```diff
@@ -0,0 +1,890 @@
+import android.os.Handler
+import android.os.HandlerThread
+import android.view.View
+import io.sentry.SentryLevel
+import io.sentry.SentryOptions
+import io.sentry.android.replay.ScreenshotRecorderCallback
+import io.sentry.android.replay.ScreenshotRecorderConfig
+import io.sentry.android.replay.util.submitSafely
+import io.sentry.util.IntegrationUtils
+import java.util.LinkedList
+import java.util.WeakHashMap
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.LazyThreadSafetyMode.NONE
+import kotlin.math.roundToInt
+
+@SuppressLint("UseKtx")
+internal class CanvasStrategy(
+  private val executor: ScheduledExecutorService,
+  private val screenshotRecorderCallback: ScreenshotRecorderCallback?,
+  private val options: SentryOptions,
+  private val config: ScreenshotRecorderConfig,
+) : ScreenshotStrategy {
+
+  private var screenshot: Bitmap? = null
+  private val prescaledMatrix by
+    lazy(NONE) { Matrix().apply { preScale(config.scaleFactorX, config.scaleFactorY) } }
+  private val lastCaptureSuccessful = AtomicBoolean(false)
+  private val textIgnoringCanvas = TextIgnoringDelegateCanvas()
+  private var cachedPixels: IntArray? = null
+
+  private val freePictures: MutableList<PictureReaderHolder> =
+    LinkedList(
+      listOf(
+        PictureReaderHolder(config.recordingWidth, config.recordingHeight),
+        PictureReaderHolder(config.recordingWidth, config.recordingHeight),
+      )
+    )
+  private val unprocessedPictures: MutableList<PictureReaderHolder> = LinkedList()
+
+  private val processingThread = HandlerThread("screenshot_recorder.canvas")
+  private val processingHandler: Handler
+
+  init {
+    processingThread.start()
+    processingHandler = Handler(processingThread.looper)
+
+    IntegrationUtils.addIntegrationToSdkVersion("ReplayCanvasStrategy")
+  }
+
+  private val pictureRenderTask = Runnable {
+    val holder: PictureReaderHolder? =
+      synchronized(unprocessedPictures) {
+        when {
+          unprocessedPictures.isNotEmpty() -> unprocessedPictures.removeAt(0)
+          else -> null
+        }
+      }
+    if (holder == null) {
+      return@Runnable
+    }
+    try {
+      holder.reader.setOnImageAvailableListener(
+        {
+          val hwImage = it?.acquireLatestImage()
+          if (hwImage != null) {
+            val hwScreenshot = toBitmap(hwImage)
+            screenshot = hwScreenshot
+            lastCaptureSuccessful.set(true)
+            screenshotRecorderCallback?.onScreenshotRecorded(hwScreenshot)
+            hwImage.close()
+          } else {
+            options.logger.log(SentryLevel.DEBUG, "Canvas Strategy: hardware image is null")
+          }
+          synchronized(freePictures) { freePictures.add(holder) }
+        },
+        processingHandler,
+      )
+
+      val surface = holder.reader.surface
+      val canvas = surface.lockHardwareCanvas()
+      canvas.drawColor(Color.BLACK, PorterDuff.Mode.CLEAR)
```

Comment:
could we use `TRANSPARENT` here actually? Wondering if there will be some weird artifacts due to us erasing with  the black color